### PR TITLE
ROX-26528: add rpms-signature-scan task to pipeline

### PR DESCRIPTION
**Pull request by tommartensen**
## Description

Add new required task to pipeline. 

> The task should run after the build-container task and assign its IMAGE_URL and IMAGE_DIGEST results to the tasks parameters.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Updated documentation accordingly~

**Automated testing**
  - ~[ ] Added unit tests~
  - ~[ ] Added integration tests~
  - ~[ ] Added regression tests~

If any of these don't apply, please comment below.

## Testing Performed

No new tests required. Konflux CI pipeline passes.

### CODE REVIEW

File: .tekton/collector-component-pipeline.yaml [hunk 1]
```diff
@@ -563,3 +563,23 @@ spec:
     - input: $(params.skip-checks)
       operator: in
       values: [ "false" ]
+
+  - name: rpms-signature-scan
+    params:
+        - name: image-digest
+          value: $(tasks.build-image-index.results.IMAGE_DIGEST)
+        - name: image-url
+          value: $(tasks.build-image-index.results.IMAGE_URL)
```
Comment (msugakov) [inline .tekton/collector-component-pipeline.yaml:572]: [nit] The indentation looks too off here compared to other tasks.

**Review by msugakov — APPROVED** [review]
Looks legit and CI succeeded.

UI however is wonderfully broken for [this run](https://console.redhat.com/application-pipeline/workspaces/rh-acs/applications/acs/pipelineruns/collector-build-bch8n/detail) and I only see `No logs found.` for every task.

image

I don't want to block the review on this as we can address issues in subsequent PRs.

**Comment by tommartensen** [comment]
> Looks legit and CI succeeded.
> UI however is wonderfully broken for [this run](https://console.redhat.com/application-pipeline/workspaces/rh-acs/applications/acs/pipelineruns/collector-build-bch8n/detail) and I only see No logs found. for every task.

There were no warnings or errors reported by the scan when I looked at the results.

**Comment by tommartensen** [comment]
This is the output for the task (from the second commit in this PR). The task iterates through all architectures found in the index image to check for the presence of unsigned RPMs. 

[collector-build-9pcrc-rpms-signature-scan.log](https://github.com/user-attachments/files/17375401/collector-build-9pcrc-rpms-signature-scan.log)

**Pull request merged**